xray/backends/pynio_.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+from .. import Variable
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin
+from ..core import indexing
+
+from .common import AbstractDataStore
+
+
+class NioArrayWrapper(NDArrayMixin):
+    def __init__(self, array, ds):
+        self.array = array
+        self._ds = ds  # make an explicit reference because pynio uses weakrefs
+
+    @property
+    def dtype(self):
+        return np.dtype(self.array.typecode())
+
+    def __getitem__(self, key):
+        if key == () and self.ndim == 0:
+            return self.array.get_value()
```

Comment:
It would be nice to support indexing with a scalar array like `array[()]` or `array[...]` in pynio. This pretty well in NumPy, and means you don't need special case methods like `get_value`.